django/db/migrations/state.py

```diff
@@ -352,7 +414,10 @@ def update_model_field_relation(
         remote_model_key = resolve_relation(model, *model_key)
         if remote_model_key[0] not in self.real_apps and remote_model_key in concretes:
             remote_model_key = concretes[remote_model_key]
-        self.relations[remote_model_key][model_key].append((field_name, field))
+        if field_name not in self.models[model_key].fields.keys():
```

Comment:
Handling the positive case first would look cleaner:

```python
if field_name in self.models[model_key].fields:
```

Note also that I don't think you need to use `keys()`.